### PR TITLE
add `with_overrides` to map task

### PR DESCRIPTION
**Pull request by samhita-alla**
Signed-off-by: Samhita Alla <aallasamhita@gmail.com>

# TL;DR
`with_overrides` is not covered in the map task API. 

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
https://github.com/flyteorg/flyte/issues/1574

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/core/map_task.py [hunk 1]
```diff
@@ -230,6 +230,11 @@ def map_task(task_function: PythonFunctionTask, concurrency: int = None, min_suc
         all inputs are processed.
     :param min_success_ratio: If specified, this determines the minimum fraction of total jobs which can complete
         successfully before terminating this task and marking it successful.
+
+    ``with_overrides`` on a map task can be used to set individual map task resource assigment.
```
Comment (samhita-alla) [inline flytekit/core/map_task.py:234]: @katrogan, what's the preferred way of assigning resourced to individual map tasks–`TaskMetadata`/`Resources` or `with_overrides`?

Comment (katrogan) [reply flytekit/core/map_task.py:234]: I believe it's with `with_overrides` but I haven't worked on flytekit in so long I'm not sure if it's changed! cc @wild-endeavor @eapolinario

Comment (wild-endeavor) [reply flytekit/core/map_task.py:234]: could you try this out? I believe this should work.   calling `map_task(t1)` produces the map task, and then `()` on that should produce a `Node` which should hit https://github.com/flyteorg/flytekit/blob/master/flytekit/core/node.py#L79

Comment (wild-endeavor) [reply flytekit/core/map_task.py:234]: but note that this is on the node... not the map task itself.  if you use the map task twice, you have to call with overrides twice.

Comment (wild-endeavor) [reply flytekit/core/map_task.py:234]: let me know when you want a +1

Comment (samhita-alla) [reply flytekit/core/map_task.py]: `TaskMetadata` works, whereas `Resources` doesn't. I replaced `Resources` with `with_overrides` in the docstring and test case.

**Review by wild-endeavor — APPROVED** [review]

**Pull request merged**